app/controllers/AgentInternationalAddressController.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.annotations.EstateRegistration
+import controllers.actions._
+import forms.AgentInternationalAddressFormProvider
+import javax.inject.Inject
+import models.{Mode, NormalMode}
+import navigation.Navigator
+import pages.{AgentInternationalAddressPage, AgentNamePage}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import utils.countryOptions.CountryOptionsNonUK
+import views.html.AgentInternationalAddressView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AgentInternationalAddressController @Inject()(
+                                      override val messagesApi: MessagesApi,
+                                      sessionRepository: SessionRepository,
+                                      @EstateRegistration navigator: Navigator,
+                                      identify: IdentifierAction,
+                                      getData: DataRetrievalAction,
+                                      requireData: DataRequiredAction,
+                                      requiredAnswer: RequiredAnswerActionProvider,
+                                      formProvider: AgentInternationalAddressFormProvider,
+                                      val controllerComponents: MessagesControllerComponents,
+                                      view: AgentInternationalAddressView,
+                                      val countryOptions: CountryOptionsNonUK
+                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  private def actions() =
+    identify andThen getData andThen requireData andThen
```

Comment:
Same as previous comment, about giving a name to this sequence.

Reply:
Updated